File: CSA2/sourse files/flower.cpp
//------------------------------------------------------------------------------
// flower.cpp - содержит реализацию класса flower
//------------------------------------------------------------------------------

#include "flower.h"
#include <stdexcept>

//------------------------------------------------------------------------------
// Преобразование int к flower type
Flower::type Flower::IntToFlowerType(int value) {
	switch (value) {
	case Flower::type::HOME:
		return Flower::type::HOME;
	case Flower::type::WILD:
		return Flower::type::WILD;
	case Flower::type::GARDEN:
		return Flower::type::GARDEN;
	default:
		throw std::invalid_argument("Invalid flower type");
	}
}

//------------------------------------------------------------------------------
// Преобразование flower type к string
string Flower::FlowerTypetoString(Flower::type value) {
	switch (value) {
	case Flower::type::HOME:
		return "HOME";
	case Flower::type::WILD:
		return "WILD";
	case Flower::type::GARDEN:
		return "GARDEN";
	default:
		throw std::invalid_argument("Invalid flower type");
	}
}

//------------------------------------------------------------------------------
// Конструктор цветка. 
Flower::Flower(std::string name, int flowerTypeNumber) :Plant(name) {
	flowerType = IntToFlowerType(flowerTypeNumber);
}

//------------------------------------------------------------------------------
// Отформатированный вывод данных. 
std::string Flower::ToString() {
	return "It is Flower: name = " + Name
		+ ", Flower type = " + FlowerTypetoString(flowerType)
		+ ". Part of Vowels = " + std::to_string(GetPartOfVowels());
};

File: CSA3/sourse files/flower.py
from plant import *
from flowerTypeEnum import *


# ----------------------------------------------
class Flower(Plant):
    flower_type = FlowerType.HOME

    def __init__(self, name, flower_type_int):
        super().__init__(name)
        self.flower_type == FlowerType(flower_type_int);

    def toString(self):
        return "It is Flower: name = {}, Flower type month = {}. Part of Vowels = {}".format(self.name,
                                                                                             self.flower_type.name,
                                                                                             self.getPartOfVowels())
